src/main/core/support/configuration.rs

```diff
@@ -174,9 +173,8 @@ pub struct GeneralOptions {
 
 impl GeneralOptions {
     /// Replace unset (`None`) values of `base` with values from `default`.
-    pub fn merge(base: &Self, default: &Self) -> Self {
-        let mut base = base.clone();
-        base.merge(default.clone());
+    pub fn merge(mut base: Self, default: Self) -> Self {
```

Comment:
> I missed this the first time, but I think it'd be more idiomatic to have these be `pub fn merge(self, default: Self) -> Self`?

The `merge()` function already exists from the `#[derive(Merge)]`, so we can't give it the name "merge". If you have other suggestions for what to name it, let me know.

> (Or maybe take `&mut self` and mutate?) Up to you

That is what the original design from the "merge" library does, but it makes the interface difficult if you want to actually mutate the second argument rather than the first. Then you need to do something like:

```rust
let temp = b.clone();
temp.merge(a);
let a = temp;
```

Reply:
If it returns a value, it'd be reasonable to name the function after what it returns. e.g. `merged_with`. I also just realized the semantics are the opposite of what I'd expect, particularly if the first parameter is some version of `self`. Maybe `overridden_with` to disambiguate, for something like `pub fn overridden_with(self, overrides: &Self) -> Self`? So at the caller it'd be something like `let settings = defaults.overriden_with(cli_args)`.

All up to you; I think the current version is ok and don't want to ask you to keep changing it :)